### PR TITLE
Migrating our Sand changes to the new version of Hydra. 

### PR DESCRIPTION
**Pull request by johnwu96822**
Changes include:
* Standard health endpoints (/health and /v1/health/detailed)
* Getting secrets and RDS cert from Secrets Manager
* Statsd metrics
* Commented out prometheus metrics, since we use statsd

- [x] @abdulchaudhrycoupa 
- [x] @supriya

### CODE REVIEW

File: cmd/server/handler.go [hunk 1]
```diff
@@ -212,7 +204,7 @@ func setup(d driver.Driver, cmd *cobra.Command) (admin *x.RouterAdmin, public *x
 	}
 
 	adminmw.Use(adminLogger)
-	adminmw.Use(d.Registry().PrometheusManager())
+	// adminmw.Use(d.Registry().PrometheusManager())
```
Comment (abdulchaudhrycoupa) [inline cmd/server/handler.go:207]: @johnwu96822 do you want to document why this is commented out?

File: cmd/server/handler.go [hunk 2]
```diff
@@ -225,73 +217,73 @@ func setup(d driver.Driver, cmd *cobra.Command) (admin *x.RouterAdmin, public *x
...
-				BatchMaxSize:         500 * 1000,
-				BatchSize:            250,
-				Interval:             time.Hour * 24,
-			},
-		},
-	)
-
-	adminmw.Use(metrics)
-	publicmw.Use(metrics)
+	// metrics := metricsx.New(
```
Comment (abdulchaudhrycoupa) [inline cmd/server/handler.go:220]: @johnwu96822 do you want to document why this is commented out?

File: health/health.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+func DetailedStatus(r InternalRegistry, c Configuration) []byte {
+	dependent := DBCheck(r)
+	status := status(dependent.State.Status, r)
+	status["project"] = GetProject()
+	status["host"] = c.IssuerURL().String()
+	status["description"] = "Sand authentication service for service to service communications."
+	status["name"] = "Sand"
+	status["uptime"] = int64(time.Since(serverStartTime).Seconds())
+	status["dependencies"] = []interface{}{dependent}
+	data, _ := json.Marshal(status)
```
Comment (abdulchaudhrycoupa) [inline health/health.go:70]: @johnwu96822 you should at least log a marshal error and panic maybe

File: cmd/server/handler.go (hunk 2 above)
Comment (johnwu96822) [reply cmd/server/handler.go:220]: Hmmm i thought I did, but I did not. Will add.

File: health/health.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	status["dependencies"] = []interface{}{dependent}
+	data, err := json.Marshal(status)
+	if err != nil {
+		data = []byte("Error producing JSON response: " + err.Error())
+	}
+	return data
+}
+
+func GetProject() projectInfo {
+	logsStr := os.Getenv("APPLICATION_LOG_LINKS")
```
Comment (supriya) [inline health/health.go:78]: are you going to set these in SM?

**Review by supriya — APPROVED** [review]

File: oauth2/handler.go [hunk 5]
```diff
@@ -551,11 +552,24 @@ func (h *Handler) TokenHandler(w http.ResponseWriter, r *http.Request) {
 	var session = NewSession("")
 	var ctx = r.Context()
 
+	// NOTE: if we can't get the accessRequest object below, then we don't know what the client_id is
+	var sandClientID, scopes string
+	if originalID, _, ok := r.BasicAuth(); ok {
+		sandClientID, _ = url.QueryUnescape(originalID)
```
Comment (supriya) [inline oauth2/handler.go:558]: do you want to handle the error here?

File: health/health.go (hunk 4 above)
Comment (supriya) [reply health/health.go:78]: I saw Sushma adding something similar once for some service. I was not sure back then if they are ever being used?

File: oauth2/handler.go (hunk 5 above)
Comment (johnwu96822) [reply oauth2/handler.go:558]: No, if error it'll just be empty string.

File: health/health.go (hunk 4 above)
Comment (johnwu96822) [reply health/health.go:78]: Yes they should be used if set in SM

**Comment by abdulchaudhrycoupa** [comment]
👍🏽

**Comment by supriya** [comment]
lgtm

**Comment by johnwu96822** [comment]
[lockit v4.9.3] merged via Lockit CLI on 2020-11-30 10:14:28.531428 -0800 PST m=+4.580455520

**Pull request merged**